File: MailWizz/Python/mailwizz_email_sender.py
"""
MailWizz Email Sender
This script provides functionality to send emails to MailWizz using the API.

Author: Rasool Vahdati
Date: 2024/10/12
License: MIT License
"""

import requests

class MailWizzEmailSender:
    def __init__(self, api_url, public_key, private_key):
        self.api_url = api_url
        self.public_key = public_key
        self.private_key = private_key

    def send_email_to_mailwizz(self, email, list_id):
        headers = {
            'X-MW-PUBLIC-KEY': self.public_key,
            'X-MW-PRIVATE-KEY': self.private_key,
            'Content-Type': 'application/json'
        }
        data = {
            'EMAIL': email
        }
        response = requests.post(f"{self.api_url}/lists/{list_id}/subscribers", json=data, headers=headers)

        return response.status_code == 200

    @staticmethod
    def validate_email(email):
        return '@' in email and '.' in email

File: MailWizz/.NET/MailWizzEmailSender.cs

// MailWizzEmailSender.cs
// Description: This class handles MailWizz email API integration in .NET (C#).
// It allows for sending email subscriptions to MailWizz via the MailWizz API.
// Author: Rasool Vahdati

using System;
using System.Net.Http;
using System.Text;
using System.Threading.Tasks;

public class MailWizzEmailSender
{
    private readonly string apiUrl;
    private readonly string publicKey;
    private readonly string privateKey;

    public MailWizzEmailSender(string apiUrl, string publicKey, string privateKey)
    {
        this.apiUrl = apiUrl;
        this.publicKey = publicKey;
        this.privateKey = privateKey;
    }

    // Send email to MailWizz API
    public async Task<bool> SendEmailToMailWizzAsync(string email, string listId)
    {
        using (var client = new HttpClient())
        {
            client.DefaultRequestHeaders.Add("X-MW-PUBLIC-KEY", publicKey);
            client.DefaultRequestHeaders.Add("X-MW-PRIVATE-KEY", privateKey);

            var body = new { EMAIL = email };
            var content = new StringContent(Newtonsoft.Json.JsonConvert.SerializeObject(body), Encoding.UTF8, "application/json");

            var response = await client.PostAsync($"{apiUrl}/lists/{listId}/subscribers", content);
            return response.IsSuccessStatusCode;
        }
    }

    // Validate email format
    public static bool ValidateEmail(string email)
    {
        return !string.IsNullOrEmpty(email) && email.Contains("@");
    }
}